src/algebra/direct_sum/internal.lean

```diff
@@ -159,6 +159,126 @@ begin
   simp_rw [direct_sum.coe_of_apply, ←finset.sum_filter, set_like.coe_ghas_mul],
 end
 
+lemma direct_sum.coe_of_mul_apply_eq_ite [add_left_cancel_monoid ι] [semiring R] [set_like σ R]
+  [add_submonoid_class σ R] (A : ι → σ) [set_like.graded_monoid A] {i : ι}
+  (r : A i) (r' : ⨁ i, A i) (n : ι) [decidable (∃ (j : ι), i + j = n)] :
+  ((direct_sum.of _ i r * r') n : R) =
+  if h : ∃ j, i + j = n then r * r' h.some else 0 :=
```

Comment:
Using `some` here feels very weird. Can we just use subtraction and `has_ordered_sub` somehow?

Reply:
I think this is supposed to be a general version where everything else can be derived from so that we don't have to case split `r = 0` every time.

Reply:
We've now switched to `has_ordered_sub`.